Realm/RLMApp.mm

```diff
@@ -128,21 +135,35 @@ - (instancetype)initWithBaseURL:(nullable NSString *)baseURL
         self.localAppName = localAppName;
         self.localAppVersion = localAppVersion;
         self.defaultRequestTimeoutMS = defaultRequestTimeoutMS;
-
-        _config.device_info.sdk = "Realm Swift";
-
-        // Platform info isn't available when running via `swift test`.
-        // Non-Xcode SPM builds can't build for anything but macOS, so this is
-        // probably unimportant for now and we can just report "unknown"
-        auto processInfo = [NSProcessInfo processInfo];
-        RLMNSStringToStdString(_config.device_info.platform_version,
-                               [processInfo operatingSystemVersionString] ?: @"unknown");
-        RLMNSStringToStdString(_config.device_info.sdk_version, REALM_COCOA_VERSION);
+        RLMConfigureSyncConnectionParameters(_config);
         return self;
     }
     return nil;
 }
 
+static void RLMConfigureSyncConnectionParameters(realm::app::App::Config& config) {
+    // Anonymized BundleId
+    NSString *bundleId = [[NSBundle mainBundle] bundleIdentifier];
+    NSData *bundleIdData = [bundleId dataUsingEncoding:NSUTF8StringEncoding];
+    RLMNSStringToStdString(config.device_info.bundle_id, RLMHashBase16Data(bundleIdData.bytes, bundleIdData.length));
+
+    config.device_info.sdk = "Realm Swift";
+    RLMNSStringToStdString(config.device_info.sdk_version, REALM_COCOA_VERSION);
+
+    auto processInfo = [NSProcessInfo processInfo];
+    RLMNSStringToStdString(config.device_info.platform_version,
+                           [processInfo operatingSystemVersionString] ?: @"unknown");
+
+    RLMNSStringToStdString(config.device_info.framework_version, @__clang_version__);
+
+#ifdef REALM_UIDEVICE_AVAILABLE
+    RLMNSStringToStdString(config.device_info.device_name, [UIDevice currentDevice].model);
```

Comment:
This seems like it won't work for CocoaPods or SPM installations?

Reply:
@tgoyne This is defined at the beginning of this file (on line 24) and can be replaced by `#if __has_include(<UIKit/UIDevice.h>)` if it is clearer

Reply:
The problem is that UIKit is only being linked for the framework build, so this field will not be set for the other installation methods.

Reply:
I thought linking it on the `xcconfig` file will work for all the installation method.

Reply:
I'm linking UIKit on the podspec now, tested later on macos and ios.